src/h.ts

```diff
@@ -1,5 +1,5 @@
-import { vnode, VNode, VNodeData } from './vnode';
-import * as is from './is';
+import { vnode, VNode, VNodeData } from './vnode.js';
+import * as is from './is.js';
```

Comment:
It seems that all of the source code TypeScript import paths in the source are changed from implicit/no extension to `.js`. I am assuming that this is to affect the output compiled modules' import paths to be the same for the purpose of being able to use them as ES modules in the browser.

Yes, TypeScript's support for resolution of extensions is not ES modules spec compliant. And yes, providing ES modules means that any import paths in them must have paths with extensions.

But this is not how I would do it. The source files should remain unchanged. Instead, use [this project](https://github.com/dropbox/ts-transform-import-path-rewrite) to transform the import paths conditionally. Use `ttypescript`. There's an example there that uses `ttypescript`. The `alias` object needs to be something like 
```json
{
  "^\\.(.*)(?<!\\.js)&": "$1.js"
}
```